cpp/src/arrow/c/bridge_test.cc

```diff
@@ -1872,6 +1942,13 @@ TEST_F(TestSchemaImport, Map) {
   CheckImport(expected);
 }
 
+TEST_F(TestSchemaImport, RunEndEncoded) {
+  FillPrimitive(AddChild(), "s", "run_ends");
+  FillPrimitive(AddChild(), "I", "values");
+  FillRunEndEncoded("+r");
+  CheckImport(run_end_encoded(int16(), uint32()));
```

Comment:
What happens if the first child is not one of the allowed run ends types? Can you add a test for it?
(see e.g. the `DictionaryError` test below)

Reply:
I will add a check and a test for this.

Reply:
I'm having a hard time producing a nested REE array with this API. @pitrou, what order of calls should I use to produce a `run_end_encoded(int32, run_end_encoded(int32, utf8))`?

Reply:
I figured it out now. :)